sdk/core/azure-core/CHANGELOG.md

```diff
@@ -8,29 +8,46 @@
+- Added interoperability between `std::chrono::system_clock` types and `DateTime`.
 - Added default constructor to `DateTime`.
 - Added `DateTime` supporting dates since 0001.
 - Added Base64 encoding and decoding utility APIs to the `Azure::Core` namespace available from `azure/core/base64.hpp`.
 - Added `Http::Response<void>` template specialization.
+- Added `GetHeadersAsString()` on the `Request` class.
+- Added a `platform.hpp` header file for defining commonly used OS-specific `#define` constants.
+- Added `IsCancelled()` on the `Context class.
 
 ### Breaking Changes
 
-- Removed `DateTime::operator Duration()`.
+- Removed `DateTime::operator Duration()` and `DateTime::Duration` typedef.
 - Removed `DateTime::Now()`.
 - Moved `Azure::Core::BearerTokenAuthenticationPolicy`, defined in `azure/core/credentials.hpp` to `Azure::Core::Http` namespace in `azure/core/http/policy.hpp` header.
-- Removed option `AllowBeast` from `CurlTransportSSLOptions` in `CurlTransportOptions`.
-- Changed default option `NoRevoke` from `CurlTransportSSLOptions` for the `CurlTransportOptions` to `true`. This disables the revocation list checking by default.
 - Changed type of `Token::ExpiresOn` to `DateTime`.
 - Renamed exception `OperationCanceledException` to `OperationCancelledException`.
-- Renamed methods from `Azure::Core::Context`.
-  - `IsCanceled` to `IsCancelled`
-  - `ThrowIfCanceled` to `ThrowIfCancelled`.
+- Renamed `ThrowIfCanceled()` to `ThrowIfCancelled()` on the `Azure::Core::Context` class.
 - Moved `Azure::Core::Version`, defined in `azure/core/version.hpp` to the `Azure::Core::Details` namespace.
+- Changed `Azure::Core::AuthenticationException` to derive from `std::exception` instead of `std::runtime_error`.
+- Changed the `BodyStream::Read` API from being a pure virtual function to non-virtual.
+- Removed `CurlConnection`, `CurlConnectionPool`, `CurlSession` and `CurlNetworkConnection` by making these headers private:
+  - curl_connection.hpp
+  - curl_connection_pool.hpp
+  - curl_session.hpp
+- Removed option `AllowBeast` from `CurlTransportSSLOptions` in `CurlTransportOptions`.
+- Changed default option `NoRevoke` from `CurlTransportSSLOptions` for the `CurlTransportOptions` to `true`. This disables the revocation list checking by default.
 
 ### Bug Fixes
 
 - Fixed the Curl transport adapter connection pooling when setting options.
 - Fixed setting up the default transport adapter.
+- Fixed linker error of missing pthread on Linux.
+- Initialize class data members to avoid MSVC warning.
+- Throw `Azure::Core::Http::TransportException` if `curl_easy_init()` returns a null handle.
+
+### Other changes and Improvements
+
+- Added support for distributing the C++ SDK as a source package via vcpkg.
+- Added a 3rd party JSON library (nlohmann_json) to Azure Core provided as a CMake inteface for anyone using Azure Core to be able to include a json header.
```

Comment:
Aren't we exposing a 3rd party JSON library to the end user this way, even if it is just via CMake @vhvb1989?
If we wanted to change the JSON library post-GA, wouldn't the fact that we expose `nlohmann` force us to keep exposing it to avoid a breaking change?

https://github.com/vhvb1989/azure-sdk-for-cpp/blob/977d4da1cc829e0555b8b8d7b4bd01ec4683eec4/sdk/core/azure-core/CMakeLists.txt#L69

cc @JeffreyRichter

Reply:
Nit, I wouldn't mention the JSON library since its an internal detail

Reply:
We want the JSON library to be in internal implementation detail. Unfortunately, it becomes a public thing if customers can't use a different version of the library for their own purposes because of our dependency on it.
It would be best if we could make our usage of it completely internal by modifying the source to use our own namespace or do something like this to allow us and the customer to load different versions of this JSON library side-by-side - we just repeat the code.

Reply:
> by modifying the source to use our own namespace

I like this approach.  `Azure::Core::Interna::Json`  we  wrap nlohmann src under this for the SDK services to consume